src/common/resources.rs

```diff
@@ -176,37 +185,13 @@ fn _get_cache_directory() -> PathBuf {
 /// # Example
 ///
 /// ```no_run
-/// use rust_bert::resources::{download_resource, RemoteResource, Resource};
+/// use rust_bert::resources::{RemoteResource, Resource};
 /// let model_resource = Resource::Remote(RemoteResource::from_pretrained((
 ///     "distilbert-sst2/model.ot",
 ///     "https://cdn.huggingface.co/distilbert-base-uncased-finetuned-sst-2-english-rust_model.ot",
 /// )));
-/// let local_path = download_resource(&model_resource);
+/// let local_path = model_resource.get_local_path();
 /// ```
-pub fn download_resource(resource: &Resource) -> Result<&PathBuf, RustBertError> {
-    match resource {
-        Resource::Remote(remote_resource) => {
-            let target = remote_resource.local_path.clone();
-            let url = remote_resource.url.clone();
-            if !target.exists() {
-                println!("Downloading {} to {:?}", url, target);
-                fs::create_dir_all(target.parent().unwrap())?;
-                let mut rt = Runtime::new()?;
-                let local = task::LocalSet::new();
-                local.block_on(&mut rt, async {
-                    let client = Client::new();
-                    let output_file = tokio::fs::File::create(target).await?;
-                    let mut output_file = tokio::io::BufWriter::new(output_file);
-                    let mut response = client.get(&url).send().await?;
-                    while let Some(chunk) = response.chunk().await? {
-                        output_file.write(&chunk).await?;
-                    }
-                    output_file.flush().await?;
-                    Ok::<(), RustBertError>(())
-                })?;
-            }
-            Ok(resource.get_local_path())
-        }
-        Resource::Local(_) => Ok(resource.get_local_path()),
-    }
+pub fn download_resource(resource: &Resource) -> Result<PathBuf, RustBertError> {
+    Ok(resource.get_local_path()?)
```

Comment:
just nitpicking, but I believe you could return `resource.get_local_path()` directly as this is a `Result<PathBuf, RustBertError>`